datapipelines/src/main/kotlin/io/customer/datapipelines/plugins/ScreenFilterPlugin.kt

```diff
@@ -0,0 +1,27 @@
+package io.customer.datapipelines.plugins
+
+import com.segment.analytics.kotlin.core.Analytics
+import com.segment.analytics.kotlin.core.BaseEvent
+import com.segment.analytics.kotlin.core.ScreenEvent
+import com.segment.analytics.kotlin.core.platform.EventPlugin
+import com.segment.analytics.kotlin.core.platform.Plugin
+import io.customer.datapipelines.config.ScreenView
+
+/**
+ * Plugin to filter screen events based on the configuration provided by customer app.
+ * This plugin is used to filter out screen events that should not be processed further.
+ */
+class ScreenFilterPlugin(private val screenViewUse: ScreenView) : EventPlugin {
```

Comment:
Is this meant for only internal use or are we intentionally exposing it as public API? I don't think we should commit to adding this style of filtering till we discuss this widely and how it can affect the rest of the API (filtering events and other things) and how it can play with other SDKs as well.

Reply:
Updated to internal